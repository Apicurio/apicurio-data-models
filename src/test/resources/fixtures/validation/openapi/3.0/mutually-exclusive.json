{
  "openapi": "3.0.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Petstore",
    "description": "A sample API that uses a petstore as an example to demonstrate features in the OpenAPI 3.0 specification"
  },
  "paths": {
    "/": {
      "get": {
        "operationId": "getRepositoriesByOwner",
        "responses": {
          "200": {
            "description": "Successful response"
          }
        }
      }
    }
  },
  "components": {
    "examples": {
      "ex-004": {
        "summary": "Example Summary",
        "description": "Description of example.",
        "value": "Some Value",
        "externalValue": "http://foo.bar/examples/address-example.txt"
      }
    },
    "headers": {
      "head-014": {
        "description": "The number of allowed requests in the current period",
        "schema": {
          "type": "integer"
        },
        "content": {
          "text/plain": {
            "schema": {
              "type": "string"
            }
          }
        }
      },
      "head-013": {
        "description": "The number of allowed requests in the current period",
        "example": "Some example",
        "examples": {
          "application/json": {
            "summary": "Example 1",
            "description": "An example description...",
            "value": "[1, 2, 3, 4, 5]"
          }
        }
      }
    },
    "links": {
      "link-001": {
        "operationRef": "#paths~12.0~1repositories~1{$response.body#/username}",
        "operationId": "getRepositoriesByOwner",
        "parameters": {
          "username": "$response.body#/username"
        }
      }
    },
    "parameters": {
      "mt-001": {
        "name": "fooParam",
        "in": "query",
        "description": "Some parameter desc.",
        "content": {
          "text/plain": {
            "schema": {
              "type": "string"
            },
            "example": "Quick brown fox",
            "examples": {
              "list": {
                "summary": "List of names",
                "value" : ["Bob","Diane","Mary","Bill"]
              },
              "empty":{
                "summary": "Empty List",
                "value": []
              }
            }
          }
        }
      },
      "par-030": {
        "name": "barParam",
        "in": "query",
        "description": "Some parameter desc.",
        "schema": {
          "type": "string"
        },
        "content": {
          "text/plain": {
            "schema": {
              "type": "string"
            }
          }
        }
      },
      "par-031": {
        "name": "barParam",
        "in": "query",
        "description": "Some parameter desc.",
        "schema": {
          "type": "string"
        },
        "example": "Example of value.",
        "examples": {
          "example-1": {
            "value": "blerg"
          }
        }
      }
    }
  }
}
