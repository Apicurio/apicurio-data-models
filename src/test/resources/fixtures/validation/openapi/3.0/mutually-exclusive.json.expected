[MT-001] |medium| {/components/parameters[mt-001]/content[text/plain]->example} :: Media Type "Example" and "Examples" are mutually exclusive.
[PAR-030] |medium| {/components/parameters[par-030]->schema} :: Parameter cannot have both a Schema and Content.
[PAR-031] |medium| {/components/parameters[par-031]->example} :: Parameter "Example" and "Examples" are mutually exclusive.
[EX-004] |medium| {/components/examples[ex-004]->value} :: Example "Value" and "External Value" are mutually exclusive.
[HEAD-014] |medium| {/components/headers[head-014]->schema} :: Header cannot have both a Schema and Content.
[HEAD-013] |medium| {/components/headers[head-013]->example} :: Header "Example" and "Examples" are mutually exclusive.
[LINK-001] |medium| {/components/links[link-001]->operationId} :: Link Operation Reference and Operation ID cannot both be used.